### PR TITLE
[AppService] Improved error messages for SSH Tunnel Connection

### PR DESCRIPTION
**Pull request by sanchitmehta**
**Description**  

This PR improves the error message(s) and adds fail fast scenarios for the customers using Azure CLI to SSH to their App Service on Linux instances. We leverage the new version of the status API to determine if an app container exposes the port 2222 and and provide fail fast errors with links to the documentation for customers who have incorrectly configured the SSH setup on their custom docker images.  

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by sanchitmehta** [comment]
cc: @panchagnula @btardif

File: src/azure-cli/azure/cli/command_modules/appservice/tunnel.py [hunk 1]
```diff
@@ -91,23 +92,32 @@ def is_webapp_up(self):
...
-        logger.info('Status response message: %s', msg)
-        if 'FAIL' in msg.upper():
-            logger.info('WARNING - Remote debugging may not be setup properly. Reponse content: %s', msg)
-            return False
-        if 'SUCCESS' in msg.upper():
-            return True
+        resp_msg = r.read().decode('utf-8')
+        json_data = json.loads(resp_msg)
+
+        if 'STARTED' in json_data["state"].upper():
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/tunnel.py:111]: Is it possible the key is not found

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -3467,7 +3467,7 @@ def _wait_for_webapp(tunnel_server):
         if tries == 0:
             logger.warning('Connection is not ready yet, please wait')
         if tries == 60:
-            raise CLIError("Timeout Error, Unable to establish a connection")
+            raise CLIError("SSH timeout, your app must be running before it can accept SSH connections. Use `az webapp log tail` to review the app startup logs.")
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:3470]: I don't think it's a timeout error. Maybe "Exceed maximum retry times"

File: src/azure-cli/azure/cli/command_modules/appservice/tunnel.py (hunk 1 above)
Comment (sanchitmehta) [reply src/azure-cli/azure/cli/command_modules/appservice/tunnel.py:114]: Added a check if key is not found, returning false from this method would invoke the retry logic

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (sanchitmehta) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:3470]: This is just polling for the app status (at 1 second intervals), before making an ssh tunnel connection. This message would be useful for users to understand what went wrong

**Pull request merged**